### DISCUSSION TITLE
Combine path and tags conditionals

### DISCUSSION THREAD

**Discussion by phw** [community/community#26273] [Actions]
<p>How is the logic when combining conditonals for push to restrict paths and tags or branches?</p>
<p>In my specific case I want to limit a workflow to only a specific list of paths, but I want to ensure it always runs on a tag. Interestingly if I only do something like this:</p>
<pre><code>on:
  push:
    paths:
    - 'src/**'
</code></pre>
<p>The behavior is:</p>
<ul>
<li>
<p>If a commit contains change paths in src/** the workflow runs, if it does not contain any changes in src/** it does not run. This is as expected</p>
</li>
<li>
<p>If I tag a commit that does <strong>not</strong> contain changes to src/** the workflow also runs on pushing this tag. This is kind of unexpected, but actually exactly what I want.</p>
</li>
</ul>
<p>Is my observation correct? Can I rely on the workflow being run on every tag push?</p>
<p>And what happens if I define both paths and tags, e.g:</p>
<pre><code>on:
  push:
    paths:
    - 'src/**'

    tags:

    - 'release-*'
</code></pre>
<p>The expecation would be that this runs on any push that changes src/** and also on an tag release-* independent of the paths in the changed commit. Is this correct? If not, how could I achieve this behavior?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Thank you for posting this question in Github Community forum.</p>
<p>Based on your requirement: <em>The expecation would be that this runs on any push that changes src/** and also on an tag release-* independent of the paths in the changed commit.</em></p>
<p>You can add branches filter with paths and tags together:</p>
<pre><code>on:
  push:
    paths:
      - 'src/**'
    branches:
      - master
    tags:
      - 'release-*'
</code></pre>
<p>Please note that:</p>
<p><strong>1. Path filters are not evaluated for pushes to tags</strong>. You can found this sentence <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fworkflow-syntax-for-github-actions%23onpushpull_requestpaths&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C8afd5cfcdeb14881eea608d7749a875c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637106079352424739&amp;sdata=q%2BUlpTFBhz69w75hpwKffBXTzwLOl5JQ7m4RW3sQb3M%3D&amp;reserved=0" rel="nofollow noopener">here</a>.</p>
<p>This could explain that when you use path filter only, when you tag a commit that does not contain changes to src/** the workflow also runs on pushing this tag.</p>
<p>on:</p>
<p>  push:</p>
<p>    paths:</p>
<p>    - ‘src/**’</p>
<p><strong>2. If you only define only </strong>** tags <strong><strong>or only </strong></strong> branches **<strong>, the workflow won’t run for events affecting the undefined Git ref.</strong> You can found this sentence <a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fworkflow-syntax-for-github-actions%23onpushpull_requestbranchestags&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C8afd5cfcdeb14881eea608d7749a875c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637106079352424739&amp;sdata=vvgvRxZLKbz6SIB5a95eB1CToApTwIALQLgm53%2FJtRA%3D&amp;reserved=0" rel="nofollow noopener">here</a>.</p>
<p>When you just add paths and <strong>tags</strong> filter to on push event, tags filter will stop the workflow to be triggered by pushing to branches. So I suggest you to add <strong>branches</strong> filter.</p>

**Reply by dayjaby** [reply]
Why those fishy looking links?

**Comment by phw** [comment]
<p>Thanks a lot, this fuly answers my questions. I missed that part that path filters are ignored for tag pushes, but it makes sense and really helps in my case.</p>

**Comment by chientrm** [comment]
<pre><code class="lang-yaml">name: Release CleoTown

on:
  push:
    tags:
      - "*"

jobs:
  prepare:
    runs-on: ubuntu-latest
    steps:
      - id: get_tag
        run: echo ::set-output name=TAG::${GITHUB_REF/refs\/tags\//}
      - id: cleotown
        uses: jsmith/changes-since-last-tag@v0.4.1
        with:
          glob: cleotown/**
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    outputs:
      TAG: ${{ steps.get_tag.outputs.TAG }}
      CLEOTOWN: ${{ steps.cleotown.outputs.any_changed }}

  release:
    runs-on: ubuntu-latest
    needs: prepare
    if: needs.prepare.outputs.CLEOTOWN == 'true'
</code></pre>
